xray/core/dataset.py

```diff
@@ -1028,6 +1029,58 @@ def sel(self, method=None, **indexers):
         return self.isel(**indexing.remap_label_indexers(self, indexers,
                                                          method=method))
 
+    def isel_points(self, dim='points', **indexers):
+        """Returns a new dataset with each array indexed pointwise along the
+        specified dimension(s).
+
+        This method selects pointwise values from each array and is akin to
+        the NumPy indexing behavior of `arr[[0, 1], [0, 1]]`, except this
+        method does not require knowing the order of each array's dimensions.
+
+        Parameters
+        ----------
+        dim : str, optional
+            Dimension name for which the points will be added to.
+        **indexers : {dim: indexer, ...}
+            Keyword arguments with names matching dimensions and values given
+            by integers, slice objects or arrays. All indexers must be the same
+            length.
+
+        Returns
+        -------
+        obj : Dataset
+            A new Dataset with the same contents as this dataset, except each
+            array and dimension is indexed by the appropriate indexers. In
+            general, each array's data will be a view of the array's data
+            in this dataset, unless numpy fancy indexing was triggered by using
+            an array indexer, in which case the data will be a copy.
+
+        See Also
+        --------
+        Dataset.sel
+        DataArray.isel
+        DataArray.sel
+        DataArray.isel_points
+        """
+        invalid = [k for k in indexers if k not in self.dims]
+        if invalid:
+            raise ValueError("dimensions %r do not exist" % invalid)
+
+        # all the indexers should be iterables
+        keys = indexers.keys()
+        indexers = [(k, ([v] if not isinstance(v, Sequence) else v))
```

Comment:
I would raise an error with slice objects, too. 1d arrays of integers is enough functionality.

On Wed, Jul 22, 2015 at 9:46 PM, Joe Hamman <notifications@github.com>
wrote:

>> +            an array indexer, in which case the data will be a copy.
>> +
>> +        See Also
>> +        --------
>> +        Dataset.sel
>> +        DataArray.isel
>> +        DataArray.sel
>> +        DataArray.isel_points
>> +        """
>> +        invalid = [k for k in indexers if k not in self.dims]
>> +        if invalid:
>> +            raise ValueError("dimensions %r do not exist" % invalid)
>> +
>> +        # all the indexers should be iterables
>> +        keys = indexers.keys()
>> +        indexers = [(k, ([v] if not isinstance(v, Sequence) else v))
> I've been trying to implement this and I'm not getting it to work.  Specifically, how do we want to handle `slice` objects? As I think about it more, I'm not sure how my first commit was working with `slice` objects as indexers.
> Any thoughts on the best way to handle this?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/xray/xray/pull/481/files#r35289799